docs/source/federation/metrics.md

```diff
@@ -0,0 +1,37 @@
+
+1. The Apollo gateway receives an operation from a client
+1. The Apollo gateway constructs a query plan for the operation, delegating sub-queries to underlying services
+1. For each [fetch](https://www.apollographql.com/docs/apollo-server/federation/federation-spec/#fetch-service-capabilities) to an implementing service, a response is received
+1. In the [`extensions`](https://www.apollographql.com/docs/resources/graphql-glossary/#extensions) of the response, a trace from the sub-query is exposed
+1. The gateway collects the set of sub-query traces from implementing services, and arranges them in the shape of the [query plan](https://www.apollographql.com/docs/apollo-server/federation/implementing/#inspecting-query-plans)
+1. The Federated trace is sent to the Apollo [metrics ingress](https://www.apollographql.com/docs/references/setup-analytics/#engine-reporting-endpoint) for processing.
+
+The model of federated metrics is that implementing services report timing and error information to the gateway, and the gateway is responsible for reporting those metrics.
+
+## Turning it on
+
+Ensure that all dependencies on `apollo-server` are at version `2.7.0` or higher. Provide an API key to your gateway via the `ENGINE_API_KEY` environment variable for the gateway to report metrics to the default ingress. To ensure that implementing services do not report metrics as well, do not provide them with an `ENGINE_API_KEY` or set `{ engine: true, reporting: false }` in the constructor options to ApolloServer. These options will cause the Apollo gateway to collect tracing information from the underlying federated services and pass them on, along with the query plan, to the Apollo metrics ingress. To change which endpoint you would like to report the tracing information to, you can modify the `endpointUrl` in the `engine` key of the constructor to Apollo Server. Currently, only Apollo Server supports detailed metrics insights as an implementing service, but we would love to work with you to implement the protocol in other languages!
+
+> NOTE: By default, metrics will be reported to the `'current'` variant. To change the variant for reporting, set the `ENGINE_GRAPH_VARIANT` environment variable.
+
+## How tracing data is exposed from a federated service
+
+The Apollo gateway looks to the `extensions` field of all service responses for the presence of an `ftv1` field. This field contains a representation of the tracing information for the sub-query that was executed against the service, sent as the base64 encoding of the [protobuf representation](https://github.com/apollographql/apollo-server/blob/master/packages/apollo-engine-reporting-protobuf/src/reports.proto) of the trace. By default, a federated Apollo Server service will emit this tracing information in its extension if sent the header pair `'apollo-federation-include-trace': 'ftv1'`. By setting the gateway's API key via the environment variable, it will send this header pair by default as well.
```

Comment:
Nit capitalized Base64.

```suggestion
The Apollo gateway looks to the `extensions` field of all service responses for the presence of an `ftv1` field. This field contains a representation of the tracing information for the sub-query that was executed against the service, sent as the Base64 encoding of the [protobuf representation](https://github.com/apollographql/apollo-server/blob/master/packages/apollo-engine-reporting-protobuf/src/reports.proto) of the trace. By default, a federated Apollo Server service will emit this tracing information in its extension if sent the header pair `'apollo-federation-include-trace': 'ftv1'`. By setting the gateway's API key via the environment variable, it will send this header pair by default as well.
```

Reply:
This makes things sound like a lot of additional work is necessary when the answer is that they need not do _anything_.

> By default, a federated Apollo Server service will emit this tracing information in its extension if sent the header pair `'apollo-federation-include-trace': 'ftv1'`. By setting the gateway's API key via the environment variable, it will send this header pair by default as well.

Also, I suspect it doesn't _have_ to be set via environment variable, but that `engine: { apiKey: 'set' }` is enough?